File: core/server/project/commonBase/src/main/java/com/home/commonBase/scene/base/UnitLogicBase.java
package com.home.commonBase.scene.base;

import com.home.commonBase.data.scene.unit.UnitData;

/** 单位逻辑体基类 */
public class UnitLogicBase extends SceneObjectLogicBase
{
	protected UnitData _data;
	
	protected Unit _unit;
	
	@Override
	public void setObject(SceneObject obj)
	{
		super.setObject(obj);
		
		_unit=(Unit)obj;
	}

	@Override
	public void construct()
	{
		
	}

	@Override
	public void init()
	{
		super.init();
		_data=_unit.getUnitData();
	}

	@Override
	public void dispose()
	{
		super.dispose();
		_data=null;
	}
	
	/** 每份时间 */
	public void onPiece(int delay)
	{
		
	}
	
	
}

File: core/client/game/src/commonGame/scene/base/UnitLogicBase.cs
using System;
using ShineEngine;

/// <summary>
/// 单位逻辑体基类
/// </summary>
public class UnitLogicBase:SceneObjectLogicBase
{
	protected UnitData _data;

	protected Unit _unit;

	public UnitLogicBase()
	{
	}

	public override void setObject(SceneObject obj)
	{
		base.setObject(obj);

		_unit=obj as Unit;
	}

	/** 初始化 */
	public override void init()
	{
		base.init();
		_data=_unit.getUnitData();
	}

	/** 析构 */
	public override void dispose()
	{
		base.dispose();

		_data=null;
	}

	/** 每秒十次 */
	public virtual void onPiece(int delay)
	{

	}
}